### PR TITLE
Add/republicize preview

### PR DESCRIPTION
**Pull request by jhnstn**
This adds a preview modal to publicize and updates share action button styles.

**Action buttons**

`breakpoint > 480px`
screen shot 2017-04-03 at 1 41 57 pm

`breakpoint < 480px`
action-buttons

**Modal**
`breakpoint > 660px `

screen shot 2017-03-30 at 1 28 36 pm

`breakpoint < 660px`

screen shot 2017-03-30 at 1 29 05 pm

**Note** 
At the moment we only have previews for Facebook and Twitter. #12644 adds the remaining available services.

**Testing**
- build with `ENABLE_FEATURES=publicize/preview make run`
- Open sharing option on a post from http://calypso.localhost:3000/posts/my/:site
- Edit customized message
- Click "Preview"
- The preview modal should open
- The preview should show the customized message
- Click the close modal button ( top left X )
- The preview modal should close

### CODE REVIEW

File: client/my-sites/post-share/sharing-preview-modal.jsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	} = props;
+
+	return (
+		<Dialog isVisible={ isVisible } additionalClassNames="post-share__sharing-preview-modal" >
+			<header className="post-share__sharing-preview-modal-header">
+				<button onClick={ onClose } className="post-share__sharing-preview-modal-close" >
+					<Gridicon icon="cross" />
+				</button>
+			</header>
+			<SharingPreviewPane { ...previewProps } />
```
Comment (gwwar) [inline client/my-sites/post-share/sharing-preview-modal.jsx]: Copying over notes from the other PR:

This is optional, but if we're concerned with pass through props, perhaps:
```jsx
{ children }
```

So in the caller this looks like:
```jsx
<SharingPreviewModal isVisible={ isVisible } onClose={ onClose }>
    <SharingPreviewPane siteId={ siteId } postId={ postId } message={ message } />
</SharingPreviewModal>
```

Comment (gwwar) [inline client/my-sites/post-share/sharing-preview-modal.jsx]: If we pull out the sharing specific props, this component becomes reusable if we need a dialog that looks like a web preview. (We'd need to change class names though and move this into the component folder.)

Comment (retrofox) [inline client/my-sites/post-share/sharing-preview-modal.jsx]: yup, it's a good idea. I think we should move this component into its folder straight in this PR.

Comment (gwwar) [inline client/my-sites/post-share/sharing-preview-modal.jsx]: I don't mind waiting for follow up PR to do this change though

**Comment by retrofox** [comment]
Looks pretty good Jason.





I wonder if we should consider the case when the user has one more account of the same media.
In the screenshot above I have two twitter accounts. I think it could/should be defined after this implementation.

**Comment by gwwar** [comment]
> I wonder if we should consider the case when the user has one more account of the same media.

For a first pass I think taking the first enabled account works.

File: client/my-sites/post-share/sharing-preview-modal.jsx (hunk 1 above)
Comment (jhnstn) [inline client/my-sites/post-share/sharing-preview-modal.jsx]: I'll land this now and do the follow up extraction in another PR.

**Pull request merged**